app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user (not logged in)
+
+    if user.admin?
+      can :manage, :all
+    else
+      can :read, :all
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
```

Comment:
- [ ] What you imply here is that a user can destroy his post and comment, but can not update or view any, it will be nice to refactor your code like this ` can :manage, Post, autho_idr: user_id` here a user can view, update and destroy his post and comment